### PR TITLE
Store all moderation history

### PR DESCRIPTION
**Pull request by dracos**
This allows moderation_original_data to store all moderation changes, and shows them on admin report /update page.
Fixes https://github.com/mysociety/keepitinthecommunity/issues/121

### CODE REVIEW

**Comment by dracos** [comment]
Added test for db storage, adding test for admin now (which through up obvious bug with just a smoke test so that's nice), here's what it looks like on admin page:
screen shot 2018-11-23 at 09 55 09

File: templates/web/base/admin/report_edit.html [hunk 1]
```diff
@@ -182,6 +182,29 @@
...
+            [% IF diff.coords %]<br>[% loc('Latitude/Longitude:') %] [% diff.coords %][% END %]
+            [% IF diff.category %]<br>[% loc('Category:') %] [% diff.category %][% END %]
+            [% IF diff.extra %]<br>[% loc('Extra data:') %] [% diff.extra %][% END %]
+        </li>
+        [% last_history = history %]
+      [% END %]
+    </ul>
+</div>
+[% END %]
+
```
[the remark this answers is not in the document]
Comment (dracos) [reply templates/web/base/admin/report_edit.html]: Have added a commit that associates the admin log entry with the moderation object row, not the problem/update; which seems to work okay, not sure if there's some reason not to do that  I haven't thought of, or some other issue...

**Comment by dracos** [comment]
Adding the reason seemed straightforward enough, hopefully done, let me know about the downgrade.

File: db/downgrade_0065---0064.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+BEGIN;
+
+ALTER TABLE admin_log DROP CONSTRAINT admin_log_object_type_check;
+
+ALTER TABLE admin_log ADD CONSTRAINT admin_log_object_type_check CHECK (
+    object_type = 'problem'
+    OR object_type = 'update'
+    OR object_type = 'user'
+);
```
[the remark this answers is not in the document]
Comment (struan) [reply db/downgrade_0065---0064.sql]: Yes, I think in practice it's unlikely to be used so dropping them all is much easier.

**Comment by struan** [comment]
Yay for reason. And refactor is nice too.

**Pull request merged**